File: search/binary_search/python/binary_search.py
def binary_search(list, num):
    found = False
    first = 0
    last = len(list) - 1
    while first <= last and not found:
        middle = (first + last)//2
        if list[middle] == num:
            found = True
        elif list[middle] < num:
            first = middle + 1
        else:
            last = middle - 1

    return found


def test():
    list = [2, 3, 7, 8, 78, 99, 102, 5555]
    item = 3
    assert binary_search(list, item) == True


if __name__ == '__main__':
    test()

File: search/binary_search/Matlab/binary_search.m
array = input('\nNumber of array:');

initializeArray = zeros(array);

for loop = 1:array
    initializeArray(loop) = input(sprintf('\nInput number [%d]:', loop));
end

item = input('\nEnter the number to be search:');

first = 1;
last = array;

while first <= last
    middle = round((first + last) / 2);
    if(initializeArray(middle) == item)
        fprintf('\nThe number is found in index: %d', middle)
        break
    elseif(initializeArray(middle) < item)
        first = middle + 1;
    elseif(initializeArray(middle) > item)
        last = middle - 1;
    end
end

if(first > last)
    fprintf('\nItem not found')
end